presto-spi/src/main/java/com/facebook/presto/spi/block/MapBlockBuilder.java

```diff
@@ -467,6 +470,10 @@ private static void buildHashTableStrict(
                     throw new RuntimeException(throwable);
                 }
 
+                if (isDuplicateKey == null) {
+                    throw new PrestoException(NOT_SUPPORTED, "map key cannot be indeterminate");
```

Comment:
ditto about "indeterminate" in error messages.

Reply:
@martint @haozhun 

How about

`map key cannot be null or contain nulls` ?